src/eq_fault_geom/geomio/cfm_shp_to_vtk.py

```diff
@@ -0,0 +1,214 @@
+    """
+    num_cells = num_trace_points - 1
+    cell_array = np.zeros((num_cells, 4), dtype=np.int)
+    for cell_num in range(num_cells):
+        cell_array[cell_num,0] = cell_num
+        cell_array[cell_num,1] = cell_num + 1
+        cell_array[cell_num,2] = cell_num + num_trace_points + 1
+        cell_array[cell_num,3] = cell_num + num_trace_points
+
+    cells = [("quad", cell_array)]
+
+    return cells
+
+
+def create_triangles(num_trace_points):
+    """
+    Create quad cells given the number of points at the surface.
+    """
+    num_quads = num_trace_points - 1
+    num_triangles = 2*num_quads
+    cell_array = np.zeros((num_triangles, 3), dtype=np.int)
+    cell_num = 0
+    for quad_num in range(num_quads):
+        cell_array[cell_num,0] = quad_num
+        cell_array[cell_num,1] = quad_num + 1
+        cell_array[cell_num,2] = quad_num + num_trace_points + 1
+        cell_num += 1
+        cell_array[cell_num,0] = quad_num + num_trace_points + 1
+        cell_array[cell_num,1] = quad_num + num_trace_points
+        cell_array[cell_num,2] = quad_num
+        cell_num += 1
+
+    cells = [("triangle", cell_array)]
+
+    return cells
+    
+    
+def create_stirling_vtk(fault_info: pd.Series, section_id: int, nztm_geometry: LineString):
+    """
+    Create 3D Stirling fault file from 2D map info.
+    """
+    # Get dip rotation matrix and create mesh from surface info.
+    dip_dir = fault_info["Dip_Dir"]
+    dip_rotation = calculate_dip_rotation(nztm_geometry, dip_dir)
+    mesh = create_mesh_from_trace(fault_info, nztm_geometry, dip_rotation)
+
+    # Write mesh.
+    file_name = fault_info["FZ_Name"].replace(" ", "_")
+    # Note this only works for faults numbered 1-9.
+    if (file_name[-1].isnumeric()):
+        file_list = list(file_name)
+        file_list[-1] = num_replace_dict[file_name[-1]]
+        file_name = "".join(file_list)
+    file_path = Path(file_name)
+    output_file = Path.joinpath(output_path, file_path).with_suffix(vtk_suffix)
+    meshio.write(output_file, mesh, file_format="vtk", binary=False)
+
+    return
+
+if __name__ == '__main__':
+    # Example file; should work on whole dataset too
+    shp_file = "../../../data/cfm_shapefile/cfm_lower_n_island.shp"
+    
+    # read in data
+    shp_df = gpd.GeoDataFrame.from_file(shp_file)
+
+    # Sort alphabetically by name
+    sorted_df = shp_df.sort_values("Name")
+
+    # Reset index to line up with alphabetical sorting
+    sorted_df = sorted_df.reset_index(drop=True)
+
+    # Reproject traces into lon lat
+    sorted_wgs = sorted_df.to_crs(epsg=4326)
+
+    # Loop through faults, creating a VTK file for each.
+    for i, fault in sorted_wgs.iterrows():
+        # Extract NZTM line for dip direction calculation/could be done in a better way, I'm sure
+        nztm_geometry_i = sorted_df.iloc[i].geometry
+
+        # Create Stirling fault segment and write VTK file.
+        create_stirling_vtk(fault, section_id=i, nztm_geometry=nztm_geometry_i)
```

Comment:
I think if the function returns a mesh, then all you need to pass in is the fault object? It has geometry, and the id isn't needed)

Reply:
This has been changed.